### DISCUSSION TITLE
Next router - persist query string changes

### DISCUSSION THREAD

**Discussion by penx** [storybookjs/storybook#24936] [Help]
### Summary

I have a toggle component that stores state in the query string, using the nextjs@12 pages router.

I can mount the component in Storybook, and see "nextRouter.push" events in the Actions panel when I click the Toggle, but the state of the Toggle remains the same.

I guess storybook is not writing these changes to the route. Is there any way to mock this? Particularly query string/search values, so that the components are interactive in storybook without having to write separate controller code for our app vs our storybook?

I understand Storybook uses the query string internally, but perhaps a mocked nextjs query string could be stored in memory?

Using
- @storybook/react@7.5.3
- @storybook/nextjs@7.5.3

**Comment by sonamoman** [comment]
Having the same issue. Especially in a search related component it is common practice to store the state as a query parameter in the url using router.push (useRouter()) and then retrieve that query parameter elsewhere using router.query.paramName.
So in my case i can too see the action of router.push but router.query.paramName never changes. Is there a workaround for that?

Using

- @storybook/react@7.4.5
- @storybook/nextjs@7.4.5
